pkg/controllers/clusterinventory/clusterprofile/suite_test.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+package clusterprofile
+
+import (
+	"context"
+	"flag"
+	"path/filepath"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog/v2"
+	"k8s.io/klog/v2/textlogger"
+	clusterinventory "sigs.k8s.io/cluster-inventory-api/apis/v1alpha1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/metrics/server"
+
+	clusterv1beta1 "go.goms.io/fleet/apis/cluster/v1beta1"
+)
+
+var (
+	cfg       *rest.Config
+	mgr       manager.Manager
+	k8sClient client.Client
+	testEnv   *envtest.Environment
+	ctx       context.Context
+	cancel    context.CancelFunc
+)
+
+func TestAPIs(t *testing.T) {
+	RegisterFailHandler(Fail)
+
+	RunSpecs(t, "ClusterProfile Controller Suite")
+}
+
+var _ = BeforeSuite(func() {
+	ctx, cancel = context.WithCancel(context.TODO())
```

Comment:
Probably gonna need the setLogger methond to silence the complaint from controller runtime 😣.

Reply:
where should the setLogger be called?

Reply:
Hi Ryan! I think it can be called any place before the controller manager starts